Dockerfile

```
@@ -1,112 +1,163 @@
-# Uses Ubuntu as base
-FROM ubuntu:24.04
+# syntax=docker/dockerfile:1.20
+FROM ubuntu:24.04 AS base
+
+RUN --mount=type=cache,target=/var/cache/apt,id=fuzz-apt-cache-base \
+    --mount=type=cache,target=/var/lib/apt/lists,id=fuzz-apt-lists-base \
+    apt-get update && apt-get install -y --no-install-recommends --no-install-suggests \
+    libpython3-dev \
+    libsodium-dev \
+    openjdk-21-jre-headless
```

We could also add llvm-dev to the stage base.

This will fix the symbolizer error since the final stage will have the [llvm-symbolizer](https://llvm.org/docs/CommandGuide/llvm-symbolizer.html) 
```suggestion
    openjdk-21-jre-headless \
    llvm-dev
```

---

I added but still wont create pretty stack traces `/usr/bin/llvm-symbolizer-18: error: 'linux-vdso.so.1': No such file or directory`. The images are back at 700mbs already. I think we can skip this

---

It's working on my end. You can see function names and line numbers now (`decode_n /build/external/lightning/common/bolt11.c:320`). The extra MB seems worth it for useful stack traces and avoiding a confusing error on startup.
```
deserialize_invoice-1  | INFO: found LLVMFuzzerCustomMutator (0x5646265b3c10). Disabling -len_control by default.
deserialize_invoice-1  | INFO: Running with entropic power schedule (0xFF, 100).
deserialize_invoice-1  | INFO: Seed: 2351783278
deserialize_invoice-1  | INFO: Loaded 2 modules   (182587 inline 8-bit counters): 145557 [0x564627bfdb08, 0x564627c2139d), 37030 [0x56462893ad10, 0x564628943db6), 
deserialize_invoice-1  | INFO: Loaded 2 PC tables (182587 PCs): 145557 [0x564627c213a0,0x564627e59cf0), 37030 [0x10c000090000,0x10c000120a60), 
deserialize_invoice-1  | INFO:      313 files found in /app/data/deserialize_invoice/corpus
deserialize_invoice-1  | === BitcoinFuzz Module Activation Log ===
deserialize_invoice-1  | Target: deserialize_invoice
deserialize_invoice-1  | Loaded Modules (6):
deserialize_invoice-1  |   [1] Lnd
deserialize_invoice-1  |   [2] Ldk
deserialize_invoice-1  |   [3] NLightning
deserialize_invoice-1  |   [4] CLightning
deserialize_invoice-1  |   [5] Eclair
deserialize_invoice-1  |   [6] LightningKmp
deserialize_invoice-1  | Active Custom Mutators:
deserialize_invoice-1  |   - BOLT11 Bech32 Custom Mutator
deserialize_invoice-1  | =========================================
deserialize_invoice-1  | INFO: seed corpus: files: 313 min: 4b max: 210b total: 35365b rss: 232Mb
deserialize_invoice-1  | #314   INITED cov: 2741 ft: 4998 corp: 238/26Kb exec/s: 314 rss: 571Mb
deserialize_invoice-1  | #315   NEW    cov: 2741 ft: 4999 corp: 239/26Kb lim: 210 exec/s: 315 rss: 571Mb L: 177/210 MS: 1 Custom-
deserialize_invoice-1  | #344   NEW    cov: 2741 ft: 5000 corp: 240/27Kb lim: 210 exec/s: 344 rss: 571Mb L: 187/210 MS: 8 CrossOver-Custom-ShuffleBytes-Custom-CMP-Custom-InsertRepeatedBytes-Custom- DE: "\001\000\000\000\000\000\000\220"-
deserialize_invoice-1  |        NEW_FUNC[1/1]: 0x5646270dcf60 in decode_n /build/external/lightning/common/bolt11.c:320
deserialize_invoice-1  | #622   NEW    cov: 2750 ft: 5013 corp: 241/27Kb lim: 210 exec/s: 622 rss: 571Mb L: 203/210 MS: 4 CustomCrossOver-CMP-Custom-CustomCrossOver- DE: " - "-
```

---

We ended up copying the symbolizer to the final image